src/repr-test-util/src/lib.rs

```diff
@@ -47,58 +47,60 @@ where
     })
 }
 
-/// Constructs a `Datum` from `litval` and `littyp`.
+/// Constructs a `Row` from a sequence of `litval` and `littyp`.
 ///
 /// See [get_scalar_type_or_default] for creating a `ScalarType`.
 ///
-/// Because Datums do not own their strings, if `littyp` is
-/// `ScalarType::String`, make sure that `litval` has already
-/// been unquoted by [unquote_string].
-///
-/// Generally, `litval` can be parsed into a Datum in the manner you would
+/// Generally, each `litval` can be parsed into a Datum in the manner you would
 /// imagine. Exceptions:
 /// * A Timestamp should be in the format `"\"%Y-%m-%d %H:%M:%S%.f\""` or
 ///   `"\"%Y-%m-%d %H:%M:%S\""`
-pub fn get_datum_from_str<'a>(litval: &'a str, littyp: &ScalarType) -> Result<Datum<'a>, String> {
-    if litval == "null" {
-        return Ok(Datum::Null);
-    }
-    match littyp {
-        ScalarType::Bool => Ok(Datum::from(parse_litval::<bool>(litval, "bool")?)),
-        ScalarType::Numeric { .. } => Ok(Datum::from(parse_litval::<Numeric>(litval, "Numeric")?)),
-        ScalarType::Int16 => Ok(Datum::from(parse_litval::<i16>(litval, "i16")?)),
-        ScalarType::Int32 => Ok(Datum::from(parse_litval::<i32>(litval, "i32")?)),
-        ScalarType::Int64 => Ok(Datum::from(parse_litval::<i64>(litval, "i64")?)),
-        ScalarType::Float32 => Ok(Datum::from(parse_litval::<f32>(litval, "f32")?)),
-        ScalarType::Float64 => Ok(Datum::from(parse_litval::<f64>(litval, "f64")?)),
-        ScalarType::String => Ok(Datum::from(litval)),
-        ScalarType::Timestamp => {
-            let datetime = if litval.contains('.') {
-                NaiveDateTime::parse_from_str(litval, "\"%Y-%m-%d %H:%M:%S%.f\"")
-            } else {
-                NaiveDateTime::parse_from_str(litval, "\"%Y-%m-%d %H:%M:%S\"")
-            };
-            Ok(Datum::from(datetime.map_err(|e| {
-                format!("Error while parsing NaiveDateTime: {}", e)
-            })?))
+///
+/// Not all types are supported yet. Currently supported types:
+/// * string, bool, timestamp
+/// * all flavors of numeric types
+pub fn test_spec_to_row<'a, I>(datum_iter: I) -> Result<Row, String>
```

Comment:
Did you want to add another repr_test_tuil/tests/testdata file with some examples of rows? Those have been really useful for me in figuring out what the input to these functions looks like

Reply:
Sure! I added `src/repr-test-util/tests/testdata/row`.